src/App.tsx

```diff
@@ -1,26 +1,259 @@
+    return todos.reduce((count, todo) => {
+      if (todo.id !== 0 && !todo.completed) {
+        return count + 1;
+      }
+
+      return count;
+    }, 0);
+  }, [todos]);
+
+  useEffect(() => {
+    getTodos()
+      .then(todosPromise => {
+        setTodos(todosPromise);
+      })
+      .catch(() => setError(ErrorMessages.UnableToLoad));
+  }, []);
+
+  useEffect(() => {
+    if (error !== ErrorMessages.Empty) {
+      const timer = setTimeout(() => {
+        setError(ErrorMessages.Empty);
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+
+    return;
+  }, [error]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  function handleKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
+    setError(ErrorMessages.Empty);
+    if (event.key !== 'Enter') {
+      return;
+    }
+
+    event.preventDefault();
+    if (inputTodo.trim() === '') {
+      setError(ErrorMessages.TitleNotBeEmpty);
+    } else {
+      setLoading(true);
+      setTodos(currentTodos => [
+        ...currentTodos,
+        {
+          id: 0,
+          title: inputTodo,
+          userId: USER_ID,
+          completed: false,
+        },
+      ]);
+      createTodo({
+        title: inputTodo.trim(),
+        userId: USER_ID,
+        completed: false,
+      })
+        .then(newTodo => {
+          setTodos(currentTodos => {
+            currentTodos.pop();
+
+            return [...currentTodos, newTodo];
+          });
+        })
+        .then(() => setInputTodo(''))
+        .catch(() => {
+          setTodos(currentTodo => {
+            currentTodo.pop();
+
+            return [...currentTodo];
+          });
+          setError(ErrorMessages.UnableAddATodo);
+        })
+        .finally(() => {
+          setLoading(false);
+        });
+    }
+  }
+
+  function handleEveryCompleted(todosSelect: Todo[]) {
+    setError(ErrorMessages.Empty);
+    setLoading(true);
+    const newCompletedState = !todosSelect.every(td => td.completed);
+
+    todosSelect
+      .filter(td => td.completed !== newCompletedState)
+      .forEach(todo => {
+        updateTodo({
+          id: todo.id,
+          userId: USER_ID,
+          title: todo.title,
+          completed: newCompletedState,
+        })
+          .then(updatedTodo => {
+            setTodos(state =>
+              handleThenUpdateCompleted(state, todo, updatedTodo),
+            );
+          })
+          .catch(() => setError(ErrorMessages.UnableUpdate))
+          .finally(() => setLoading(false));
+      });
+  }
+
+  function handleDeleteCompletedTodos() {
+    setError(ErrorMessages.Empty);
+    setLoading(true);
+    todos
+      .filter(el => el.completed)
+      .forEach(todoCompleted => {
+        deleteTodo(todoCompleted.id)
+          .then(() => {
+            setTodos(tds => tds.filter(el => el.id !== todoCompleted.id));
+          })
+          .catch(() => setError(ErrorMessages.UnableDelete))
+          .finally(() => setLoading(false));
```

Comment:
Could use `Promise.all` to `setLoading(false)` only after all delete requests are completed